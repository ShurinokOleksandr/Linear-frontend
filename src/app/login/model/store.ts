import { userStore } from '@/provider/userStore/userStore';
import { userStoreType } from '@/provider/userStore';
import { makeAutoObservable } from 'mobx';

class LoginStore {
	usernameValidationError = '';
	passwordValidationError= '';
	username = '';
	password = '';
	userStore  ;
	constructor(userStore: userStoreType) {
		makeAutoObservable(this);
		this.userStore = userStore;
	}
	async  submitLogin(){
		if(this.password.length >= 6 && this.username){
			this.setUsernameValidationError('');
			this.setPasswordValidationError('');
			// try {
			// 	const result = await api.post('/auth/login',{
			// 		username:this.username,
			// 		password:this.password
			// 	});
			// 	this.userStore.setUser(result.data);
			// 	console.log(result.data);
			// }catch (e) {
 			// 	if (axios.isAxiosError(e)){
			// 		 if(e?.response?.status === 404){
			// 			 return this.setUsernameValidationError(e?.response?.data.message);
 			// 		 }
			// 	    if(e?.response?.status === 403){
			// 		    return this.setPasswordValidationError(e?.response?.data.message);
			// 	    }
			//
			// 	}
			//
			// }
			try {
				const result =  await fetch('http://localhost:4001/auth/login',{
					body:JSON.stringify({
						username:this.username,
						password:this.password
					}),
					headers:{
						'Content-type': 'application/json',
					},
					credentials: 'include',
					method:'POST'
				});
				const data = await result.json();
				if(result.ok){
					this.userStore.setUser(data);
					return result;
				}
				if(result.status === 404){
					return this.setUsernameValidationError(data.message);
				}
				if(result.status === 403){
					return this.setPasswordValidationError(data.message);
				}
				
			}catch (e) {
				if (e instanceof Error){
					console.log(e);
				}
				
			}
		}
		if(!this.username.length) {
			this.setUsernameValidationError('Please enter an username for login.');
		}
		if(this.password.length < 6) {
			this.setPasswordValidationError('Length of password should be minimum 6 letters');
		}
	}
	setPassword(password:string){
		this.password = password;
		if(this.password.length >= 6){
 			this.setPasswordValidationError('');
		}
	}
	setUsername(username:string) {
		this.username = username;
		this.setUsernameValidationError('');
	}
	setUsernameValidationError(error:string){
		this.usernameValidationError = error ;
	}
	setPasswordValidationError(error:string){
		this.passwordValidationError = error ;
	}
	
}
export default new LoginStore(userStore);