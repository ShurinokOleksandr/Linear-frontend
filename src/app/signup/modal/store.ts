import { userStoreType, userStore } from '@/provider/userStore';
import { makeAutoObservable } from 'mobx';

class SignUpStore {
	repeatPasswordValidationError = '';
	usernameValidationError = '';
	passwordValidationError= '';
	repeatPassword = '';
	loading = false;
	username = '';
	password = '';
	userStore  ;
	constructor(userStore: userStoreType) {
		makeAutoObservable(this);
		this.userStore = userStore;
	}
	async submitSignUp(){
		if(this.password.length >= 6 && this.username && this.password === this.repeatPassword){
			this.loading = true;
			this.setUsernameValidationError('');
			this.setPasswordValidationError('');
			const result = await fetch('http://localhost:4001/auth/create',{
				body:JSON.stringify({
					repeatPassword:this.repeatPassword,
					username:this.username,
					password:this.password
				}),
				headers:{
					'Content-type': 'application/json',
				},
				credentials: 'include',
				method:'POST'
			});
			const data = await result.json();
			if(result.ok){
				this.loading = false;
				this.userStore.setUser(data);
				return data;
			}
			if(result.status === 405){
				this.loading = false;
				return this.setUsernameValidationError(data.message);
			}
			
		}
		if(!this.username.length) {
			this.setUsernameValidationError('Please enter an username for login.');
		}
		if(this.password.length < 6 ) {
			this.setPasswordValidationError('Length of password should be minimum 6 letters');
		}
		if( this.password !== this.repeatPassword) {
			this.setRepeatPasswordValidationError('Passwords are not equals');
		}
	}
	setPassword(password:string){
		this.password = password.trim();
		if(this.password.length >= 6){
			this.setPasswordValidationError('');
		}
		if(this.password.length === this.repeatPassword.length && this.password === this.repeatPassword){
			this.setRepeatPasswordValidationError('');
		}
		if(this.password !== this.repeatPassword) {
			this.setRepeatPasswordValidationError('Passwords are not equals');
		}
	}
	setRepeatPassword(repeat:string){
		this.repeatPassword = repeat.trim();
		
		if(this.password.length === this.repeatPassword.length && this.password === this.repeatPassword){
			this.setRepeatPasswordValidationError('');
		}
		
		if(this.password !== this.repeatPassword) {
			this.setRepeatPasswordValidationError('Passwords are not equals');
		}
	}
	setUsername(username:string) {
		this.username = username.trim();
		this.setUsernameValidationError('');
	}
	setRepeatPasswordValidationError(error:string){
		this.repeatPasswordValidationError = error ;
	}
	setUsernameValidationError(error:string){
		this.usernameValidationError = error ;
	}
	setPasswordValidationError(error:string){
		this.passwordValidationError = error ;
	}
	
}
export const signUpStore = new SignUpStore(userStore);