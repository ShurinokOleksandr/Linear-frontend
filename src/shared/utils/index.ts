export { externalApi } from './api/axiosInstance';
export type { UserType } from './types/types';

 