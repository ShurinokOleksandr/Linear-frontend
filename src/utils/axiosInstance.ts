import axios from 'axios';
import ky from 'ky';

export const BASE_URL = 'http://localhost:4001';
export const api = axios.create({
	headers:{
		'Content-Type':'application/json'
	},
	withCredentials:true,
	baseURL:BASE_URL
});







export const Api = ky.create({
	headers:{
		'Content-type':'application/json'
	},
	credentials:'include',
	prefixUrl:BASE_URL
}
);
// const access_token = cookies().get('access_token');
// const refresh_token = cookies().get('refresh_token');

// api.interceptors.request.use((config) => {
// 	config.headers.Authorization = `Bearer ${access_token}`;
// 	return config;
// });
// api.interceptors.response.use((config) => {
// 	return config;
// },
// async (error) =>{
//
// 	const originalRequest = error.config;
// 	if(error.response.status === 401 && error.config &&  !error.config._isRetry){
// 		originalRequest._isRetry = true;
// 		try {
// 			const response = await api.post('/auth/refresh',{ refresh_token });
// 			console.log(response);
// 			localStorage.setItem('token', response.data);
// 			return api.request(originalRequest);
//
// 		} catch (e) {
// 			console.log('НЕ АВТОРИЗОВАН');
// 		}
// 	}
// 	throw error;
// }
// );