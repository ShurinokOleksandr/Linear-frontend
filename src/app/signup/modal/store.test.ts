import { describe, expect, test } from '@jest/globals';
import { mockFetch } from '@/shared/utils/mock-fetch';
import 'isomorphic-fetch';

import { signUpStore } from './store';


describe('Testing sign up store',() => {
	it('Set username', () => {
		signUpStore.setUsername('sar');
		expect(signUpStore.username).toBe('sar');
	});
	it('Set password', () => {
		signUpStore.setPassword('sar');
		expect(signUpStore.password).toBe('sar');
	});
	it('Set repeat password', () => {
		signUpStore.setRepeatPassword('sar');
		expect(signUpStore.repeatPassword).toBe('sar');
	});
	it('Expect error not equals passwords', () => {
		signUpStore.setPassword('sar1');
		signUpStore.setRepeatPassword('sar');
		expect(signUpStore.repeatPasswordValidationError).toBe('Passwords are not equals');
	});
	test('Sign Up',async () => {
		signUpStore.setUsername('tt');
		window.fetch = mockFetch(JSON.stringify({ name:signUpStore.username,id:111 }));
		signUpStore.setPassword('111111');
		signUpStore.setRepeatPassword('111111');
		const result = await signUpStore.submitSignUp();
		expect(signUpStore.repeatPassword).toBe('111111');
		expect(signUpStore.password).toBe('111111');
		expect(signUpStore.username).toBe('tt');
		expect(result).toEqual(JSON.stringify({ name:'tt',id:111 }));
	});
	it('Expect error user exist',async () => {
		signUpStore.setUsername('sa');
		signUpStore.setPassword('111111');
		signUpStore.setRepeatPassword('111111');
		await signUpStore.submitSignUp();
		expect(signUpStore.password).toBe('111111');
		expect(signUpStore.repeatPassword).toBe('111111');
		expect(signUpStore.username).toBe('sa');
		expect(signUpStore.usernameValidationError).toBe('User with this name already exist');
	});
	it('Testing create user',async () => {
		signUpStore.setUsername('testUsasdsadasasddasdae1r');
		signUpStore.setPassword('111111');
		signUpStore.setRepeatPassword('111111');
		const result = await signUpStore.submitSignUp();
		console.log(result);
		expect(signUpStore.password).toBe('111111');
		expect(signUpStore.repeatPassword).toBe('111111');
		expect(signUpStore.username).toBe('testUsasdsadasasddasdae1r');
		expect(result).toMatchObject({ name:signUpStore.username });
	});
	it('Testing non empty username',async () => {
		signUpStore.setUsername('');
		signUpStore.setPassword('111111');
		signUpStore.setRepeatPassword('111111');
		await signUpStore.submitSignUp();
		expect(signUpStore.password).toBe('111111');
		expect(signUpStore.repeatPassword).toBe('111111');
		expect(signUpStore.username).toBe('');
		expect(signUpStore.usernameValidationError).toBe('Please enter an username for login.');
	});
	it('Testing equals passwords',async () => {
		signUpStore.setUsername('sa');
		signUpStore.setPassword('111111');
		signUpStore.setRepeatPassword('1111111');
		await signUpStore.submitSignUp();
		expect(signUpStore.repeatPassword).toBe('1111111');
		expect(signUpStore.password).toBe('111111');
		expect(signUpStore.username).toBe('sa');
		expect(signUpStore.repeatPasswordValidationError).toBe('Passwords are not equals');
	});
});