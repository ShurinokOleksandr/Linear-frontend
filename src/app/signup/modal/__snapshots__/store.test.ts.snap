// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing sign up store Testing create user 1`] = `
{
  "id": 25,
  "name": "testUser",
  "success": 201,
}
`;
